src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetWrapperBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResourceSet"/> and the <see cref="ODataResource"/>'s that are part of it.
+    /// </summary>        
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1012:AbstractTypesShouldNotHaveConstructors")]
+    public abstract class ODataResourceSetWrapperBase : ODataItemBase
+    {
+        /// <summary>
+        /// To determint the type of Resource Set
+        /// </summary>        
+        internal abstract ResourceSetType ResourceSetType { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataResourceSetWrapper"/>.
+        /// </summary>
+        /// <param name="item">The wrapped item.</param>
+        public ODataResourceSetWrapperBase(ODataResourceSetBase item)
+            : base(item)
+        {
+            Resources = new List<ODataResourceWrapper>();
+        }
+
+        /// <summary>
+        /// Gets the wrapped <see cref="ResourceSetBase"/>.
+        /// </summary>
+        public ODataResourceSetBase ResourceSetBase
+        {
+            get
+            {
+                return Item as ODataResourceSetBase;
+            }
+        }
+
+        /// <summary>
+        /// Gets the members of this ResourceSet.
+        /// </summary>
+        public IList<ODataResourceWrapper> Resources { get; }
```

Comment:
>public IList<ODataResourceWrapper> Resources { get; } [](start = 8, length = 53)

Later, if we support DeltaLink, DeltaDeletedLink, How can we expand this list? 
At that time, it will be a breaking change again? @mikepizzo

Reply:
We are saying that we will not support DeltaLink, DeltaDeletedLink in the 7.x codebase -- just a not supported scenario.  This needs to be baked into the 8.x codebase, where we can make breaking changes, before we RTM.

Reply:
In the typeless scenario, we support the DeltaLink and DeltaDeletedLink (for serializer).

So, In my opinion, we should support (at least):
1) have the typed class DeltaLink<T>, etc.
2) can read and create DeltaLink<T>
This can support the basic scenario (to read and writer).

For how to patch link, i think it's a feature, I do think we should not support to patch link in 7.x.

Reply:
There isn't a clean way to deserialize deltalink/deltadeletedlink in a ODataResourceSetWrapper without making breaking changes.

Using the 4.01 nested @delta is the preferred way to represent an updated relationship in a delta payload, so the deltedLink/deltaDeletedLink is a much lower priority scenario.We can not fix this for 7.x.

---
In reply to: [602073865](https://github.com/OData/WebApi/pull/2428#discussion_r602073865) [](ancestors = 602073865)